engine/src/main/java/org/terasology/rendering/world/WorldRenderer.java

```diff
@@ -33,6 +34,13 @@
  * the update and render methods as they are central to a rendering implementation.
  */
 public interface WorldRenderer {
+    // This is exposed so that controller-related functions can create callbacks.
+    // Ideally, these classes would be able to do so with dependency injection, but that
+    // is broken for at least FirstPersonHeldMountPointComponent.java. Could be reltaed to:
+    // https://github.com/MovingBlocks/Terasology/issues/2336A
+    // TODO: take this out and do dependency injection properly for any classes that depend on this being exposed.
+    public OpenVRProvider VR_PROVIDER = new OpenVRProvider();
```

Comment:
Yes, I confirm this is pretty ugly. :smile: 

So, in FirstPersonHeldItemMountPointComponent you can get the WorldRender via @In but not the OpenVRProvider?